plugins/rbac/src/components/RolesList/RolesList.tsx

```diff
@@ -24,6 +25,14 @@ const useStyles = makeStyles(theme => ({
 export const RolesList = () => {
   const { toastMessage, setToastMessage } = useToast();
   const { openDialog, setOpenDialog, deleteRoleName } = useDeleteDialog();
+  const location = useLocation();
+
+  useEffect(() => {
+    if (location?.state?.toastMessage) {
+      setToastMessage(location.state.toastMessage);
+    }
+    return () => setToastMessage('');
+  }, [location, setToastMessage]);
```

Comment:
this seems repeated https://github.com/janus-idp/backstage-plugins/pull/1354/files#diff-ee9611ae8a736eb0f13552c9aef1d0591064074dda9f3628bad2e4ad08449f7bR17-R24

so does this make sense to move this to custom hook?

Reply:
Good point, I have updated this PR to use a custom hook instead of repeating this useEffect block.